lib/absinthe/phase/schema/field_imports.ex

```diff
@@ -29,11 +29,14 @@ defmodule Absinthe.Phase.Schema.FieldImports do
     Schema.InputObjectTypeDefinition,
     Schema.InterfaceTypeDefinition
   ]
+  @exclude_fields [
+    :__typename
+  ]
   def import_fields(%def_type{} = type, types) when def_type in @can_import do
     Enum.reduce(type.imports, type, fn {source, opts}, type ->
       source_type = Map.fetch!(types, source)
 
-      rejections = Keyword.get(opts, :except, [])
+      rejections = Keyword.get(opts, :except, []) ++ @exclude_fields
```

Comment:
I chose to always exclude `__typename` since the current behavior was actually causing it to be included twice for `object` as well as once for `input_object`